### PR TITLE
Update CONTRIBUTING.md and DEVELOPMENT_WORKFLOW.md files

### PR DESCRIPTION
**Pull request by avatarjohn**
## Description

This PR updates the CONTRIBUTING.md and DEVELOPMENT_WORKFLOW.md files in the Bittensor project.

## Changes Made

- In CONTRIBUTING.md,

  - Add `Getting Started` Section
  - Add `Peer Review` Section
  - Create new doc file named 'Code_Review_doc' from CONTRIBUTING.md file. 

- In DEVELOPMENT_WORKFLOW.md, 
  - Add several sections about Bittensor project's overview.
  - Add some parts of general style and ways to develop based on Bitcoin developer guidline.

### CODE REVIEW

File: contrib/CONTRIBUTING.md [hunk 1]
```diff
@@ -34,6 +38,35 @@ Additionally, note that the core implementation of Bittensor consists of two sep
 
 Supplemental, yet necessary repositories are [openvalidators](https://github.com/opentensor/validators) and [openminers](https://github.com/opentensor/miners) which contain Bittensor Validators and Miners (respectively) designed by the OpenTensor Foundation team and open-sourced for the community to use.
 
+## Getting Started
+New contributors are very welcome and needed.
+Reviewing and testing is highly valued and the most effective way you can contribute as a new contributor. It also will teach you much more about the code and process than opening pull requests. 
+
+Before you start contributing, familiarize yourself with the Bittensor Core build system and tests. Refer to the documentation in the repository on how to build Bittensor core and how to run the unit tests, functional tests, and fuzz tests.
```
Comment (ifrit98) [inline contrib/CONTRIBUTING.md]: Small nit: Does bittensor have fuzz tests?  This looks directly copped from bitcoin. 

It's fine (and encouraged) to use bitcoin as a reference, we are standing on the shoulders of giants, but please make sure to refer only to things that actually exist in bittensor to not introduce ambiguity.

File: contrib/CONTRIBUTING.md [hunk 2]
```diff
@@ -34,6 +38,35 @@ Additionally, note that the core implementation of Bittensor consists of two sep
...
+
+There are many open issues of varying difficulty waiting to be fixed. If you're looking for somewhere to start contributing, check out the [good first issue](https://github.com/opentensor/bittensor/issues?page=3&q=is%3Aissue+is%3Aclosed) list or changes that are up for grabs. Some of them might no longer be applicable. So if you are interested, but unsure, you might want to leave a comment on the issue first.
```
Comment (ifrit98) [inline contrib/CONTRIBUTING.md]: This link is pointed to closed issues. Is this intended?

Comment (ifrit98) [reply contrib/CONTRIBUTING.md]:
```suggestion
There are many open issues of varying difficulty waiting to be fixed. If you're looking for somewhere to start contributing, check out the [good first issue](https://github.com/opentensor/bittensor/labels/good%20first%20issue) list or changes that are up for grabs. Some of them might no longer be applicable. So if you are interested, but unsure, you might want to leave a comment on the issue first.
```

File: contrib/CONTRIBUTING.md [hunk 3]
```diff
@@ -34,6 +38,35 @@ Additionally, note that the core implementation of Bittensor consists of two sep
...
+### Good First Issue Label
+The purpose of the good first issue label is to highlight which issues are suitable for a new contributor without a deep understanding of the codebase.
+
+However, good first issues can be solved by anyone. If they remain unsolved for a longer time, a frequent contributor might address them.
+
+You do not need to request permission to start working on an issue. However, you are encouraged to leave a comment if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.
+### Beginner and Help-wanted Issues Label
+You can start by looking through these `beginner` and `help-wanted` issues:
+
+* Beginner issues - issues which should only require a few lines of code, and a test or two.
```
Comment (ifrit98) [inline contrib/CONTRIBUTING.md]: Created label.  It's nice to link to it for clarity.
```suggestion
* [Beginner issues](https://github.com/opentensor/bittensor/labels/beginner) - issues which should only require a few lines of code, and a test or two.
```

File: contrib/CONTRIBUTING.md [hunk 4]
```diff
@@ -34,6 +38,35 @@ Additionally, note that the core implementation of Bittensor consists of two sep
...
+* Help wanted issues - issues which should be a bit more involved than `beginner` issues.
```
Comment (ifrit98) [inline contrib/CONTRIBUTING.md]: Ditto here

```suggestion
* [Help wanted issues](https://github.com/opentensor/bittensor/labels/help%20wanted) - issues which should be a bit more involved than `beginner` issues.
```

File: contrib/CONTRIBUTING.md [hunk 5]
```diff
@@ -34,6 +38,35 @@ Additionally, note that the core implementation of Bittensor consists of two sep
...
+
+Both issue lists are sorted by total number of comments. While not perfect, number of comments is a reasonable proxy for impact a given change will have.
```
Comment (ifrit98) [inline contrib/CONTRIBUTING.md]: This is not the case.  Make sure we're molding this to bittensor and not just copy-pasting from bitcoin.

Comment (ifrit98) [inline contrib/CONTRIBUTING.md]:
```suggestion
```

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 6]
```diff
@@ -1,63 +1,410 @@
...
+        - [Security](#security)
+        - [Release Verification](#release-verification)
+        - [Post-release Actions](#post-release-actions)
+    - [Notes for Logging](#notes-for-logging)
+        - [Logging Setup](#logging-setup)
+        - [Logging Messages](#logging-messages)
+        - [Logging Variables](#logging-variables)
+        - [Logging Exceptions](#logging-exceptions)
+
+# 1. Introduction
```
Comment (ifrit98) [inline contrib/DEVELOPMENT_WORKFLOW.md]: This is a lot.  Seems better suited for docs.


What is the relationship to development flow here?

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 7]
```diff
@@ -1,63 +1,410 @@
...
+
+### Subtensor
+
+Subtensor is a lower-level protocol that handles communication between neurons. It ensures that data is transfered securely and efficiently across the network.
+
+### Metagraph
+
+The Metagraph is a decentralized ledger that keeps track of the state of the Bittensor network. It records which neurons are part of the network and how they are connected.
+
+# 4. Preliminary Developer Notes
```
Comment (ifrit98) [inline contrib/DEVELOPMENT_WORKFLOW.md]: Ditto.  Nice information, but how does this relate to development workflow?  Is likely better suited documentation material.

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 8]
```diff
@@ -1,63 +1,410 @@
...
+
+Project Structure
+-------------------
+Typically, a project may have the following structure:
+
+- Root Directory: Contains configuration files, README, LICENSE, and other metadata files.
```
Comment (ifrit98) [inline contrib/DEVELOPMENT_WORKFLOW.md]: This is just copy-pasta.  No. 

Please don't do this because bittesnor does NOT have `/src`

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 9]
```diff
@@ -1,63 +1,410 @@
...
+
+- `Generators:` Use generators when dealing with large amounts of data to save memory.
+
+- `Context Managers:` Use context managers (with statement) for resource management.
+
+- `String Formatting:` Use f-strings for formatting strings in Python 3.6 and above.
+
+- `Error Handling:` Use exceptions for error handling whenever possible.
+
+### 4. Documentation
```
Comment (ifrit98) [inline contrib/DEVELOPMENT_WORKFLOW.md]: Better suited to style guide.

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 10]
```diff
@@ -1,63 +1,410 @@
...
+
+`sync`: This method syncs the metagraph with the chain. It can use either a lite version or a full version of the neurons in the chain.
+
+`save`: This method saves the state of the metagraph to a file.
+
+`load`: This method loads the state of the metagraph from a file.
+
+`load_from_path`: This method loads the state of the metagraph from a file at a given path.
+
+#### Properties
```
Comment (ifrit98) [inline contrib/DEVELOPMENT_WORKFLOW.md]: What does this have to do with development workflow?

File: contrib/DEVELOPMENT_WORKFLOW.md [hunk 11]
```diff
@@ -1,63 +1,410 @@
...
+`from_scale_encoding Function`: This function is used to decode data from SCALE format. It uses the `scalecodec` library to handle the decoding process. The function takes a list of integers (the serialized data), the type of data, and flags indicating whether the data is a vector or an option. It returns a dictionary representing the decoded data.
+
+`Data Classes`: The script defines several data classes (`NeuronInfo`, `NeuronInfoLite`, `PrometheusInfo`, `DelegateInfo`, `SubnetInfo`) to represent different types of information in the system. Each class includes a `from_vec_u8` method to create an instance of the class from serialized data, and a `fix_decoded_values` method to correct the format of decoded data.
+
+`ProposalVoteData and ProposalCallData`: These are data structures used for handling proposal data in the system. `ProposalVoteData` is a dictionary that includes information about a proposal's votes, while `ProposalCallData` is a generic call object from the `scalecodec` library.
+
+Please note that this script requires the bittensor and scalecodec libraries, which should be installed and properly configured in your environment.
+
+
+Notes for Commit
```
Comment (ifrit98) [inline contrib/DEVELOPMENT_WORKFLOW.md]: This is clearly outlined in the new STYLE.md guide.  Unnecessarily duplicated and would be effective to just refer/link to the guide instead of slightly different guidelines.

**Review by ifrit98 — CHANGES_REQUESTED** [review]
Overall, lots of good nuggets in here, but also some superfluous material that can be removed or moved into more appropriate areas (such as documentation rather than development flow).

Specifically, I don't agree with lines 48-349 of development_workflow.md. These can safely be removed and would increase clarity and modularity, imo.

Great job, and thanks for opening a PR! However, needs work and some trimming before is ready.

File: contrib/DEVELOPMENT_WORKFLOW.md (hunk 9 above)
Comment (avatarjohn) [reply contrib/DEVELOPMENT_WORKFLOW.md]: All Python script style section or only 4th Documentation is better suited?
